### PR TITLE
remove max drain duration

### PR DESCRIPTION
**Pull request by ramaraochavali**
Some protocols needs more drain duration it seems https://github.com/istio/istio/discussions/50477. Envoy does not have max - So not sure if we should have one

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pkg/config/validation/validation.go [hunk 1]
```diff
@@ -1390,10 +1390,6 @@ func ValidateDrainDuration(drainTime *durationpb.Duration) (errs error) {
 			errors.New("drain time only supports durations to seconds precision"))
 	}
 
-	if drainDuration > drainTimeMax {
```
Comment (hzxuzhonghu) [inline pkg/config/validation/validation.go:1393]: Why it was there?

If  remove, drainTimeMax should be removed too

Comment (ramaraochavali) [reply pkg/config/validation/validation.go:1393]: Do not the know the history. But seems not needed. Removed the var also

**Review by howardjohn — APPROVED** [review]
LGTM

I wouldn't recommend setting it to 100hr personally... but if someone wants to :shrug:

**Comment by ramaraochavali** [comment]
> I wouldn't recommend setting it to 100hr personally... but if someone wants to 🤷

Same here

**Pull request merged**